projects/showcase/cypress/support/e2e-driver.ts

```diff
@@ -1,198 +1,179 @@
-      return this;
-    },
-    waitForMapLoaded: (timeout: number = 30000): E2eDriver => {
-      cy.get('[data-loaded="true"]', { timeout: timeout }).should('exist');
-      return this;
-    },
-    clickLanguageButton: (language: string): E2eDriver => {
-      cy.get('.lang-button').contains(language).click();
-      return this;
-    },
-    clickPopupCloseButton: (): E2eDriver => {
-      cy.get('.maplibregl-popup-close-button').click();
-      return this;
-    },
-    clickFromCodeRadioButton: (): E2eDriver => {
-      cy.get('mat-radio-button').contains('from code').click();
-      return this;
-    },
-    clickStreetsRadioButton: (): E2eDriver => {
-      cy.get('mat-radio-button').contains('streets').click();
-      return this;
-    },
-    clickCountryNamesButton: (): E2eDriver => {
-      cy.get('mat-button-toggle').contains('countries names').click();
-      return this;
-    },
-    clickZoomToBoundsButton: (): E2eDriver => {
-      cy.get('.zoom-button').click();
-      return this;
-    },
-    clickHelloCustomButton: (): E2eDriver => {
-      cy.on('window:alert', this.alertStub);
-      cy.get('.custom-control').click();
-      return this;
-    },
-    clickEnableTerrainControlButton: (): E2eDriver => {
-      cy.get('.maplibregl-ctrl-terrain').click();
-      return this;
-    },
-    clickDisableTerrainControlButton: (): E2eDriver => {
-      cy.get('.maplibregl-ctrl-terrain-enabled').click();
-      return this;
-    },
-    clickHideControlsButton: (): E2eDriver => {
-      cy.get('button').contains('Hide Controls').click();
-      return this;
-    },
+  given = {
+    alertStub: () => cy.on("window:alert", this.helper.given.stub("alert")),
   };
+  when = {
+    wait: (ms: number) => this.helper.when.wait(ms),
 
-  assert = {
-    isSameAsSnapshot: (): E2eDriver => {
-      this.compareToReference().should('eq', 0);
-      return this;
-    },
-    isNotSameAsSnapshot: (): E2eDriver => {
-      this.compareToReference().should('be.gt', 0);
-      return this;
-    },
-    mapCanvasExists: (): E2eDriver => {
-      cy.get('canvas.maplibregl-canvas').should('have.length', 1);
-      return this;
-    },
-    mapObjectLoaded: (): E2eDriver => {
-      cy.get('[data-loaded="true"]').should('exist');
-      return this;
-    },
-    mapTerrainButtonIsOff: (): E2eDriver => {
-      cy.get('.maplibregl-ctrl-terrain').should('exist');
-      return this;
-    },
-    mapTerrainButtonIsOn: (): E2eDriver => {
-      cy.get('.maplibregl-ctrl-terrain-enabled').should('exist');
-      return this;
-    },
-    helloWorldPopupExists: (): E2eDriver => {
-      cy.get('.maplibregl-popup').should('exist');
-      cy.get('.custom-popup-class1').should('exist');
-      cy.get('.custom-popup-class2').should('exist');
-      cy.get('.maplibregl-popup-content').within(() => {
-        cy.get('div').should('have.text', 'Hello world !');
+    resetConsoleWarnings: () => {
+      cy.get("@consoleWarnSpy").then((spy: any) => {
```

Comment:
According to your architecture, there should not be calls to `cy` in the driver right?

Reply:
Yes, not quite sure what to do with this one, as well as the two commands that are in the commands file
https://github.com/maplibre/ngx-maplibre-gl/blob/3ea724c8ea03999a3ad83c3d9fba778faa5f64b9/projects/showcase/cypress/support/commands.ts#L45#L53

Reply:
I moved it to MaplibreCypreeHelper